src/main/scala/mesosphere/marathon/health/HealthCheckWorkerActor.scala

```diff
@@ -49,10 +49,10 @@ class HealthCheckWorkerActor extends Actor with ActorLogging {
   }
 
   def doCheck(task: MarathonTask, check: HealthCheck): Future[Option[HealthResult]] =
-    task.getPortsList.asScala.lift(check.portIndex) match {
-      case None =>
-        Future { Some(Unhealthy(task.getId, task.getVersion, "Invalid port index")) }
-
+    getPort(task, check) match {
+      case None => Future {
```

Comment:
If you want to improve our current code as well, you can use Future.successful and spare a thread.